assignments/lambda-classes.js

```diff
@@ -1 +1,117 @@
 // CODE here for your Lambda Classes
+
+class Person {
+    constructor(attributes) {
+    this.name = attributes.name;
+    this.age = attributes.age;
+    this.location = attributes.location;
+    this.gender =attributes.gender;
+}
+speak() {
+  return `Hello my name is ${this.name}, I am from ${this.location}`;
+}
+}
+
+class Instructor extends Person {
+    constructor(childAttributes) {
+        super(childAttributes);
+        this.specialty = childAttributes.specialty;
+        this.favLanguage = childAttributes.favLanguage;
+        this.catchPhrase = childAttributes.catchPhrase;
+    }
+    demo(subject) {
+        return `Today we are learning about ${subject}`;
+    }
+    grade(student, subject) {
+        return `${student.name} receives a perfect score on ${subject}`;
+    }
+}
+
+
+class Student extends Person {
+    constructor(gChildAttributes) {
+        super(gChildAttributes);
+        this.previousBackground = gChildAttributes.previousBackground;
+        this.className = gChildAttributes.className;
+        this.favSubjects = gChildAttributes.favSubjects;
+    }
+    listsSubjects() {
+        for (let i = 0; i < this.favSubjects.length; i++) {
+            return `${this.name} ${this.favSubjects[i]}`;
+          }
+    }
+    PRAssignment(subject) {
+        return `${figg.name} has submitted a PR for ${subject}`;
```

Comment:
you have hard-coded the student into this function - instead use the `this` keyword to access any students name that is called